plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineStandardWizardPage.java

```diff
@@ -83,11 +87,21 @@ public void modifyText(ModifyEvent event) {
     }
   }  
 
-  String getAppEngineProjectId() {
+  @Override
+  public void setPageComplete(boolean complete) {
+    if (complete) {
+      config.setAppEngineProjectId(getAppEngineProjectId());
```

Comment:
My experience with complex wizards is that they inevitably need to share information between pages.  If state isn't centralized in an object like this config object, then it leads to either (A) spreading the state out amongst the pages, which becomes unwieldy and hard to reason about (and part of the reason that it's necessary to have good identifiers for wizard pages), or (B) hosting the state on the wizard which leads to increased coupling and lots of casting.

All that said, Eclipse is somewhat inconsistent as demonstrated by the `WizardNewProjectCreationPage`, since we can't alter the project location programmatically.

I'll back this out.